libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java

```diff
@@ -157,57 +236,76 @@ public ContraIndicators getContraIndicators(VerifiableCredential vc)
             String govukSigninJourneyId, String ipAddress, String userId)
             throws CiRetrievalException {
         LOGGER.info(LogHelper.buildLogMessage("Retrieving CIs from CIMIT system"));
-
-        String payload;
-        try {
-            payload =
-                    OBJECT_MAPPER.writeValueAsString(
-                            new GetCiRequest(govukSigninJourneyId, ipAddress, userId));
-        } catch (JsonProcessingException e) {
-            throw new CiRetrievalException("Failed to serialize GetCiRequest");
-        }
-
-        var invokeRequest =
-                InvokeRequest.builder()
-                        .functionName(
-                                configService.getEnvironmentVariable(
-                                        CIMIT_GET_CONTRAINDICATORS_LAMBDA_ARN))
-                        .payload(SdkBytes.fromUtf8String(payload))
-                        .qualifier(LIVE_ALIAS)
-                        .build();
-
-        InvokeResponse response;
         try {
-            response = lambdaClient.invoke(invokeRequest);
+            if (configService.enabled(CIMIT_API_GATEWAY_ENABLED)
+                    && configService.getSecret(CIMIT_API_KEY) != null) {
+                var response = sendGetHttpRequest(userId, govukSigninJourneyId, ipAddress);
+
+                if (response.statusCode() != SC_OK) {
+                    var parsedResponse =
+                            OBJECT_MAPPER.readValue(response.body(), CimitApiResponse.class);
+                    logApiRequestError(parsedResponse);
+                    throw new CiRetrievalException(FAILED_API_REQUEST);
+                }
+                return response.body();
+            } else {
+                var payload = getPayloadForGetCiRequest(govukSigninJourneyId, ipAddress, userId);
+
+                var invokeRequest =
+                        InvokeRequest.builder()
+                                .functionName(
+                                        configService.getEnvironmentVariable(
+                                                CIMIT_GET_CONTRAINDICATORS_LAMBDA_ARN))
+                                .payload(SdkBytes.fromUtf8String(payload))
+                                .qualifier(LIVE_ALIAS)
+                                .build();
+
+                InvokeResponse response = lambdaClient.invoke(invokeRequest);
+
+                if (lambdaExecutionFailed(response)) {
+                    logLambdaExecutionError(response, CIMIT_GET_CONTRAINDICATORS_LAMBDA_ARN);
+                    throw new CiRetrievalException(FAILED_LAMBDA_MESSAGE);
+                }
+                return response.payload().asUtf8String();
+            }
         } catch (LambdaException e) {
             LOGGER.error(LogHelper.buildErrorMessage("AWSLambda client invocation failed.", e));
             throw new CiRetrievalException(FAILED_LAMBDA_MESSAGE);
+        } catch (CiRetrievalException e) {
+            throw e;
+        } catch (Exception e) {
```

Comment:
Same comments about catching `Exception` :)

Reply:
ah missed this! Thanks!